### ISSUE TITLE
Converting from provided .xlsx to CSV

### ISSUE THREAD

**Issue by maisieccino** [danielgavrilov/who-cares-its-over#1]
We just received the latest set of undergrad results today and I'd love to use this visualisation with the second year grades. Could you share how you went about converting the markbook that the department provided to CSV? Unless you did it manually with libreoffice, in which case I'd best get to work :muscle:

**Comment by danielgavrilov** [comment]
I'm just looking at it now. I think the previous years were converted to CSV directly, but this year (as you can tell) we'll have to do some manual cleanup or maybe write some very clever lines of JavaScript.

To make it easier to plug all results I prefer the "clever JavaScript" option, but since there's merging of cells and CSV doesn't support that, we would have to somehow read XSLX and I imagine it would be a pain...

Let me know if you figure something out!

**Comment by maisieccino** [comment]
This is what gets spat out for each candidate when I run it through LibreOffice's CSV converter:
```
Student Candidate Number,Student Module Level,Occurrence,,Academic Year,Programme Code,,Programme Name,Credit,0.5,0.5,0.5,0.5,0.5,,0.5,0.5,,0.5,,PP
,,,,,,,,Module Code,COMP201P,COMP202P,COMP203P,COMP204P,COMP205P,,COMP206P,COMP207P,,MSIN7008,,Pass progress to next academic year
```
Looks reasonably sensible, I think we could work with this...

Notice that it just pads out below the merged cells

**Comment by maisieccino** [comment]
But anyway, will look in more detail this evening when I have some free time. :raised_hands:

**Comment by danielgavrilov** [comment]
Yeah converting directly to CSV seems like a good idea, still some clever JavaScript required but not _too_ clever. And yes I'll have to delay this a few hours too.

**Comment by danielgavrilov** [comment]
I've done the thing you suggested (converting to CSV and then parsing with JavaScript) in a new repo: https://github.com/danielgavrilov/couldve-been-worse
